[ENH][SCHEMA] Adding an OPTIONAL _task-<label> to structural MRI acquisitions (#1185)
bids-standard/bids-specification: src/04-modality-specific-files/01-magnetic-resonance-imaging-data.md

such as `T1w` or `inplaneT1`, referenced by the defacemask image.
For example, `sub-01_mod-T1w_defacemask.nii.gz`.

> The OPTIONAL [`task-<label>`](../99-appendices/entities.md#task) entity can be used

Reviewer: ```suggestion
The OPTIONAL [`task-<label>`](../appendices/entities.md#task) entity can be used
```

Sorry. I was too quick. We also removed the prefix from the folder name, hence this doc building is still failing... 🙈

Author: Ok, maybe it's easier if I re-update my fork as well?

Reviewer: yup this may help too. 😄

in order to allow tasks during structural MR acquisitions,
for example pre-described motion paradigms such as nodding, to be described.

### RECOMMENDED metadata for tasks in anatomical imaging

<!-- This block generates a metadata table.
The definitions of these fields can be found in
  src/schema/objects/metadata.yaml
and a guide for using macros can be found at
 https://github.com/bids-standard/bids-specification/blob/master/macros_doc.md
-->
{{ MACROS___make_metadata_table(
   {
      "TaskName": "RECOMMENDED",
      "TaskDescription": "RECOMMENDED",
      "Instructions": "RECOMMENDED",
   }

> ) }}

Author: Ok, I will do that, thanks for making me aware of this

> ) }}

Reviewer: I had overlooked that too. I am used yet to "think with schema" yet.

> ) }}

Reviewer: ```suggestion
) }}

```

pretty sure we need to skip a line for the table generation macro to not start using the next paragraph as input

@tsalo seen that happen before?

such as `T1w` or `inplaneT1`, referenced by the defacemask image.
For example, `sub-01_mod-T1w_defacemask.nii.gz`.

> The OPTIONAL [`task-<label>`](../99-appendices/09-entities.md#task) entity can be used

Reviewer: ```suggestion
The OPTIONAL [`task-<label>`](../99-appendices/entities.md#task) entity can be used
```

We have removed the number prefix for those files:

https://github.com/bids-standard/bids-specification/blob/master/src/appendices/entities.md

Author: Ah this is probably a leftover from my fork of the spec not being up to date...let me also update it.

in order to allow tasks during structural MR acquisitions,
for example pre-described motion paradigms such as nodding, to be described.

> Task labels MUST be consistent across subjects and sessions.

Reviewer: Actually @melanieganz given that you are linking to the definition in the appendices, you can remove the last sentence as it is already in the definition.

@satra I have always more understood as label `task-foo` for subject 1 must **mean** the same thing as `task-foo` for subject 2. I don't think that for any other type of task based things in BIDS we have put restrictions of the type  "all participants must have done all the tasks." But the fact that you are asking the question mean we should probably clarify that in the task definition.

> Task labels MUST be consistent across subjects and sessions.

Author: Ok, then I will remove the last sentence here, but then we should make a separate PR or issue to clarify what is meant by "consistent" in the task description. I will do that as well and tag you.

> Task labels MUST be consistent across subjects and sessions.

Author: Done.

> The OPTIONAL [`task-<label>`](../99-appendices/09-entities.md#task) entity can be used
> in order to allow tasks during structural MR acquisitions, 
> for example pre-described motion paradigms such as nodding, to be described. 
> Task labels MUST be consistent across subjects and sessions.

Reviewer: ```suggestion
The OPTIONAL [`task-<label>`](../99-appendices/09-entities.md#task) entity can be used
in order to allow tasks during structural MR acquisitions,
for example pre-described motion paradigms such as nodding, to be described.
Task labels MUST be consistent across subjects and sessions.
```

Reviewer: markdown fix

Author: Thanks for fixing all the formatting, @Remi-Gau!!!

### RECOMMENDED metadata for tasks in anatomical imaging

<!-- This block generates a metadata table.
The definitions of these fields can be found in
  src/schema/objects/metadata.yaml
and a guide for using macros can be found at
 https://github.com/bids-standard/bids-specification/blob/master/macros_doc.md
-->

> {{ MACROS___make_metadata_table(
>    {
>       "TaskName": "RECOMMENDED",
>       "TaskDescription": "RECOMMENDED",
>       "Instructions": "RECOMMENDED",      
>    }
> ) }}
> 

Reviewer: ```suggestion
{{ MACROS___make_metadata_table(
   {
      "TaskName": "RECOMMENDED",
      "TaskDescription": "RECOMMENDED",
      "Instructions": "RECOMMENDED",
   }
) }}
```

Reviewer: this too should pacify the remark validation

contrast agent. The key `ContrastBolusIngredient` MAY be also be added in the
JSON file, with the same label.

> The OPTIONAL [`task-<label>`](../99-appendices/09-entities.md#task) entity can be used in order to allow tasks during structural MR acquisitions, e.g. pre-described motion paradigms such as nodding, to be described. Task labels MUST be consistent across subjects and sessions.

Reviewer: ```suggestion
The OPTIONAL [`task-<label>`](../99-appendices/09-entities.md#task) entity can be used
in order to allow tasks during structural MR acquisitions, 
for example pre-described motion paradigms such as nodding, to be described. 
Task labels MUST be consistent across subjects and sessions.
```

Reviewer: just for line breaks and to remove latin abbreviation

Reviewer: @effigies the "bad latin" workflow did not catch this case of abbreviation and it seems that the validation workflow is not running. 

Is this expected? Am I missing something?

contrast agent. The key `ContrastBolusIngredient` MAY be also be added in the
JSON file, with the same label.

> The OPTIONAL [`task-<label>`](../99-appendices/09-entities.md#task) entity can be used in order to allow tasks during structural MR acquisitions, e.g. pre-described motion paradigms such as nodding, to be described. Note, just as for functional imaging each task has a unique label that MUST only consist of letters and/or numbers (other characters, including spaces and underscores, are not allowed). Those labels MUST be consistent across subjects and sessions.

Reviewer 2: I would recommend removing the reference to the functional imaging, the label is what it is, without reference to other entries which may or may not always remain consistent.

Additionally, I think it should be highlighted that variations in acquisition quality (be they extraneous sources of noise or just head movement) are specifics of acquisition and not tasks in the context of a biological response (more information [here](https://github.com/bids-standard/bids-specification/issues/1177#issuecomment-1205817585)).

Perhaps:

"""
[...] can be used in order to document tasks performed during anatomical MR acquisition, the effects of which may pertain to brain anatomy. Note that variations in acquisition quality, as produced by either the subject or extraneous sources, are acquisition parameters and not tasks. Each task has a unique label that MUST only consist of letters and/or numbers (other characters, including spaces and underscores, are not allowed). Those labels MUST be consistent across subjects and sessions.
"""

Reviewer 3: > can be used in order to document tasks performed during anatomical MR acquisition, the effects of which may pertain to brain anatomy. Note that variations in acquisition quality, as produced by either the subject or extraneous sources, are acquisition parameters and not tasks. 

I think this definition is too limiting. And I'm not sure we should be so quick to define things such as quality; those definitions should be left up to the researcher/study as we have no idea what they're measuring.

More broadly defining tasks to be say an action/inaction (task for lack of a better word) that is performed by the subject would be my preference. Also, using `task-` in this manner allows for the use of all the accompanying features already incorporated within, e.g. events files.  Which to me makes even more of an argument for broadening task outside of `fmri`. see [here](https://bids-specification.readthedocs.io/en/stable/04-modality-specific-files/05-task-events.html)

Reviewer 4: I agree that subject actions requested by the experimenter should count as a task (e.g., resting-state scans). I still think events files should be required when the `task` entity is present though.

> The OPTIONAL [`task-<label>`](../99-appendices/09-entities.md#task) entity can be used in order to allow tasks during structural MR acquisitions, e.g. pre-described motion paradigms such as nodding, to be described. Note, just as for functional imaging each task has a unique label that MUST only consist of letters and/or numbers (other characters, including spaces and underscores, are not allowed). Those labels MUST be consistent across subjects and sessions.

Reviewer: ```suggestion
The OPTIONAL [`task-<label>`](../99-appendices/09-entities.md#task) entity can be used in order to allow tasks during structural MR acquisitions, e.g. pre-described motion paradigms such as nodding, to be described. Task labels MUST be consistent across subjects and sessions.
```

Reviewer: Remove label definition as it is already in the common principles.
